zipkin-redis/src/main/scala/com/twitter/zipkin/storage/redis/RedisIndex.scala

```diff
@@ -148,22 +148,31 @@ trait RedisIndex extends Index {
       def binaryAnnoStringify(bin: BinaryAnnotation, service: String): String =
         redisJoin(service, bin.key, encodeAnnotation(bin))
 
-      val time = span.lastAnnotation.get.timestamp
-      val binaryAnnos: Seq[Future[Unit]] = span.serviceNames.toSeq flatMap { serviceName =>
-        span.binaryAnnotations map { binaryAnno =>
-          binaryAnnotationsListMap.add(
-            binaryAnnoStringify(binaryAnno, serviceName),
-            time,
-            span.traceId
-          )
-        }
+      span.lastAnnotation.map {
```

Comment:
I think I get what you are doing here (guard on no last annotation), but it would be more clear if there was a test.